contracts/sophia.md

```diff
@@ -126,6 +126,25 @@ function root(t : tree('a)) : option('a) =
 
 *NOTE: Recursive data types are not yet implemented*
 
+### Lists
+
+A Sophia list is a dynamically sized, homogenous, immutable, singly
+linked list. A list is constructed with the syntax `[1, 2, 3]`. The
+elements of a list can be any of datatype so for example we can have
+the following lists:
+
+```
+[1, 33, 2, 666]
+[(1, "aaa"), (10, "jjj"), (666, "the beast")]
+[{ [1] = "aaa", [10] = "jjj"}, { [5] = "eee", [666] = "the beast"}]
+```
```

Comment:
It would be helpful to show the types of these lists (`list(int)`, `list((int, string))`, `list(map(int, string))`).

Reply:
Tracked as https://www.pivotaltracker.com/story/show/159792627